gremlin/ui/device.py

```diff
@@ -30,13 +31,108 @@
 from gremlin import common, event_handler, joystick_handling, shared_state
 from gremlin.error import GremlinError
 from gremlin.intermediate_output import IntermediateOutput
-from gremlin.types import InputType
+from gremlin.types import InputType, PropertyType
+from gremlin.util import file_exists_and_is_accessible
+from gremlin.common import SingletonDecorator
+from gremlin.config import Configuration
 
 
 QML_IMPORT_NAME = "Gremlin.Device"
 QML_IMPORT_MAJOR_VERSION = 1
 
 
+@SingletonDecorator
+class DeviceDatabase:
+    """Provides button/axis to name mapping for known devices"""
+
+    def __init__(self) -> None:
+        self._load()
+
+    def _load(self) -> None:
```

Comment:
Technically could be part of the constructor, though allows us to reload it in the future if needed.

Reply:
Originally I had it in the constructor, but I don't like long constructors and made it into a method. And yes, we could reload if contents changes in the future.

Reply:
Yeah I feel the same, so keeping it as is sounds good to me.